src/integration-test/java/com/commercetools/sync/integration/externalsource/types/TypeSyncIT.java

```diff
@@ -0,0 +1,712 @@
+            .hasOnlyOneElementSatisfying(message ->
+                assertThat(message).contains("Failed to update type with key 'key_1'.")
+            );
+
+        assertThat(exceptions)
+            .hasSize(1)
+            .hasOnlyOneElementSatisfying(throwable -> {
+                assertThat(throwable).isExactlyInstanceOf(CompletionException.class);
+                assertThat(throwable).hasCauseExactlyInstanceOf(ErrorResponseException.class);
+                assertThat(throwable).hasMessageContaining("Missing required value");
+            });
+
+        AssertionsForStatistics.assertThat(typeSyncStatistics).hasValues(1, 0, 0, 1);
+    }
+
+    @Test
+    public void sync_WithSeveralBatches_ShouldReturnProperStatistics() {
+        // preparation
+        // Default batch size is 50 (check TypeSyncOptionsBuilder) so we have 2 batches of 50
+        final List<TypeDraft> typeDrafts = IntStream
+                .range(0, 100)
+                .mapToObj(i -> TypeDraftBuilder.of(
+                        "key__" + Integer.toString(i),
+                        LocalizedString.ofEnglish("name__" + Integer.toString(i)),
+                        ResourceTypeIdsSetBuilder.of().addCategories().build())
+                        .description(LocalizedString.ofEnglish("description__" + Integer.toString(i)))
+                        .fieldDefinitions(singletonList(FIELD_DEFINITION_1))
+                        .build())
+                .collect(Collectors.toList());
+
+        final TypeSyncOptions typeSyncOptions = TypeSyncOptionsBuilder
+                .of(CTP_TARGET_CLIENT)
+                .build();
+
+        final TypeSync typeSync = new TypeSync(typeSyncOptions);
+
+        //test
+        final TypeSyncStatistics typeSyncStatistics = typeSync
+                .sync(typeDrafts)
+                .toCompletableFuture()
+                .join();
+
+        //assertion
+        AssertionsForStatistics.assertThat(typeSyncStatistics)
+                .hasValues(100, 100, 0, 0);
+    }
+
+    private static void assertFieldDefinitionsAreEqual(@Nonnull final List<FieldDefinition> oldFields,
+                                                       @Nonnull final List<FieldDefinition> newFields) {
+
+        IntStream.range(0, newFields.size())
+                .forEach(index -> {
+                    final FieldDefinition oldFieldDefinition = oldFields.get(index);
+                    final FieldDefinition newFieldDefinition = newFields.get(index);
+
+                    assertThat(oldFieldDefinition.getName()).isEqualTo(newFieldDefinition.getName());
+                    assertThat(oldFieldDefinition.getLabel()).isEqualTo(newFieldDefinition.getLabel());
+                    assertThat(oldFieldDefinition.getType()).isEqualTo(newFieldDefinition.getType());
+                    // no update action exists for the input hint
+                    //assertThat(oldFieldDefinition.getInputHint()).isEqualTo(newFieldDefinition.getInputHint());
```

Comment:
Please don't leave commented out lines. Since we can always get it from the history of GIT -> no reason to add more clutter to the code.